Update INSTALL

crabcrabcam opened the pull request:
I have changed the "Basic Installation" to the top as the SVN isn't the main place I believe people get the code from, and changed the Basic Install guide to have the correct instructions (with ./configure missing and instead being cmake .) and added some basic instructions for manual installation (copying the files instead of using make install) as well as a link to FreeDoom as a potential alternative if the user doesn't have iD's Doom. 

The required libraries are correct on Ubuntu and Debian systems (tested on Ubuntu 20.04) and with minor changes to packages downloaded, these instructions work on Void Linux too.

kraflab commented on prboom2/INSTALL:
```diff
@@ -74,6 +51,9 @@ Basic Installation
      can access the IWAD file.
+     
+     You can also aquire FreeDoom from https://freedoom.github.io/ which is a 
+     free replacement to the iD Doom iWADs (although if you intend to upload demos to DSDA this iWAD isn't sufficient) 
```
I think you can remove the `(although if you intend to upload demos to DSDA this iWAD isn't sufficient)` from here. That's certainly correct but it's probably not in the scope of the INSTALL doc :^)

crabcrabcam replied:
Sorted that, sorry. I forgot I wasn't committing this to the DSDA branch.

jackrjli commented on prboom2/INSTALL:
```diff
@@ -41,17 +17,15 @@ Basic Installation
-
-     Running `configure' takes awhile.  While running, it prints some
-     messages telling which features it is checking for.
+     Run `cmake .` from this folder
```
I'd recommend adding the `-DCMAKE_BUILD_TYPE=Release` flag to get a release build (cmake defaults to debug, which results in lower performance).

crabcrabcam replied:
I've made a change to this. I added a comment to say to remove it if you're wanting a debug build (I tend to run things as debug due to never quite being sure I've done everything right).

fabiangreffrath commented on prboom2/INSTALL:
```diff
@@ -41,17 +17,18 @@ Basic Installation
-     Running `configure' takes awhile.  While running, it prints some
-     messages telling which features it is checking for.
+     Run `cmake . -DCMAKE_BUILD_TYPE=Release` from this folder
+     (You should remove the -DCMAKE... if you wish to get a debug build
```
More exactly, you should provide `-DCMAKE_BUILD_TYPE=Debug` for a debug build. I would never recommend anyone to invoke CMake without any `CMAKE_BUILD_TYPE` parameter at all.

fabiangreffrath commented on prboom2/INSTALL:
```diff
@@ -41,17 +17,18 @@ Basic Installation
+     Run `cmake . -DCMAKE_BUILD_TYPE=Release` from this folder
+     (You should remove the -DCMAKE... if you wish to get a debug build
+     for better info about crashes and such if using experimental code.
+     This does reduce performance)
```
There is a stray closing parenthesis in this sentence.

fabiangreffrath commented on prboom2/INSTALL:
```diff
@@ -41,17 +17,18 @@ Basic Installation
+     for better info about crashes and such if using experimental code.
+     This does reduce performance)
+     
+     This should end with the message "Build files have been written to: [folderpath]"
```
This is missing a full stop at the end. Please write complete sentences.

fabiangreffrath commented on prboom2/INSTALL:
```diff
@@ -74,6 +54,9 @@ Basic Installation
      can access the IWAD file.
+     
+     You can also aquire FreeDoom from https://freedoom.github.io/ which is a 
+     free replacement to the iD Doom iWADs
```
This is another incomplete sentence. Also, please prefer "may" over "can".

fabiangreffrath replied:
There is still a full stop missing after this sentence.

fabiangreffrath commented on prboom2/INSTALL:
```diff
@@ -65,6 +42,9 @@ Basic Installation
      and man pages for PrBoom. If you don't have root access on the machine,
      you should ask the syadmin to do this and the next step for you.
+     
+     If you want to install manually then you must put the `prboom-plus.wad` 
```
Please prefer "should" over "must".

fabiangreffrath commented on prboom2/INSTALL:
```diff
@@ -90,6 +73,33 @@ Basic Installation
      See the README file for information about getting the Timidity
      patches needed for music support, and more.
 
+     
```
Now this part is complete obsolete and irrelevant. We should not document building of the original project here which this fork is going to replace.

fabiangreffrath replied:
Please remove the paragraph about building from SVN.

fabiangreffrath reviewed (changes requested):
Thank you very much for doing this! The documentation for this project surely needs some improvements. I request a few changes but are else fine with it.

alexey-lysiuk reviewed (commented):
I wouldn’t recommend an in-tree build. While it works without an issue at the moment, I suggest to create a new directory, and use it as a CMake build directory.

crabcrabcam commented:
> I wouldn’t recommend an in-tree build. While it works without an issue at the moment, I suggest to create a new directory, and use it as a CMake build directory.

I'm not sure how to make this one work. 

`mkdir build && cd build`
`cmake ..`
Then running `make` gives me `make: *** No targets specified and no makefile found.  Stop.` and running `make ..` gives me `make: Nothing to be done for '..'.` 

Can you give me a hand with that?

crabcrabcam commented:
@fabiangreffrath  I'm going to go through all the things you marked in this comment. 

I'm going to start with a dickish comment, please don't take it too personally. I'm not quite fond of you attacking my missed fullstops and telling me they are "incomplete sentences". I gather from your writing you are not a native English speaker, however your tone comes across as attacking someone who is trying to help. A missed fullstop is not something that requires a full comment on, but a note in another (or joining the two missed fullstop points) and perhaps instead of stating that it is "an incomplete sentence", you could word it that it's a typo, mistake, or just a small fix to make everything look slightly more professional. I say this not because I think you're wrong, but because if someone who is not perfect at English were to try to write some documentation I don't want them getting scared off by having all their grammatical mistakes pointed out in such a harsh tone. 

Other than that bit I agree with all your comments about the small changes. I can't see the missing parenthesis, but I've rewritten that sentence anyway as it was the one about debug builds. 

I've also removed the Timidity thing. I didn't know what it was, and reading the README wasn't going to hurt anyone, so I left it in. 

If you haven't seen my other comment about using a build folder, I could use some help with that as it doesn't appear to be working how I think it should (I'm not the best versed in make and C code, or code at all these days). 

Thanks for the help with this. I'm glad I could update the docs to be actually working after struggling through them to build this and failing.

fabiangreffrath commented:
@crabcrabcam I am sorry that you felt offended by my comments! Be certain that I did not mean to attack you in any way. You are right that I am not a native speaker either, so it's probably a bit hard for me to dose my tone. To me, a sentence with a missing full stop is not complete and thus, if I find one, I report an incomplete sentence. Absolutely no offense intended!

Regarding the build directory, you'll have to create it relative to the directory where the main 'CMakeList.txt` is located (i.e. `prboom2`) so that after `mkdir build && cd build` the next `cmake ..` points right there. Make sure to remove all remnants of previous cmake invocations before doing this, e.g. `CMakeCache.txt`, `CMakeFiles`, `cmake_install.cmake`, etc.

alexey-lysiuk commented:
> I'm not sure how to make this one work.
> 
> `mkdir build && cd build`
> `cmake ..`

If the current directory was the repository root initially, the second command should be `cmake ../prboom2`.

crabcrabcam commented:
> @crabcrabcam I am sorry that you felt offended by my comments! Be certain that I did not mean to attack you in any way. You are right that I am not a native speaker either, so it's probably a bit hard for me to dose my tone. To me, a sentence with a missing full stop is not complete and thus, if I find one, I report an incomplete sentence. Absolutely no offense intended!
> 
> Regarding the build directory, you'll have to create it relative to the directory where the main 'CMakeList.txt`is located (i.e.`prboom2`) so that after `mkdir build && cd build`the next`cmake ..`points right there. Make sure to remove all remnants of previous cmake invocations before doing this, e.g.`CMakeCache.txt`, `CMakeFiles`, `cmake_install.cmake`, etc.

Text is always tough with tone, sorry for the missunderstanding. I just don't want people with slightly bad English committing to a project and getting rejected for it. After all, poor English docs are better than docs that are outdated or missing. 

As for the inline build thing, deleting the cache and makefiles from the `prboom2` folder worked perfectly. It must have been seeing them and not running cmake on the main folder. Thanks for the help on that, I'll update the file now for that.

crabcrabcam commented:
There's a lot of references to the `configure` file, but I'm not sure what can be removed as most of it is also arguments for cmake. 

If someone wants that fixed can they please do it themselves as explaining to me what needs to be changed with that lot is going to be a lot harder than just doing it. 

Fingers crossed this is the last change needed and we can finally update the docs so people can compile the project :crossed_fingers:
